include/configuration/Heuristic.hpp

```diff
@@ -0,0 +1,158 @@
+//
+// This file is part of the MQT QMAP library released under the MIT license.
+// See README.md or go to https://github.com/cda-tum/qmap for more information.
+//
+
+#pragma once
+
+#include <iostream>
+
+enum class Heuristic {
+  /** maximum over all distances between any virtual qubit pair in the current
+     layer; optimizing gate-count; admissible; tight */
+  GateCountMaxDistance,
+  /** sum over all distances between any virtual qubit pair in the current
+     layer; optimizing gate-count; not admissible; tight */
+  GateCountSumDistance,
+  /** sum over all distances between any virtual qubit pair in the current layer
+     minus the upper limit of viable shared swaps; optimizing gate-count;
+     admissible; tight */
+  GateCountSumDistanceMinusSharedSwaps,
+  /** maximum of `Heuristic::GateCountMaxDistance` and
+     `Heuristic::GateCountSumDistanceMinusSharedSwaps`; optimizing gate-count;
+     admissible; tight */
+  GateCountMaxDistanceOrSumDistanceMinusSharedSwaps,
+  /** minimum cost if each virtual qubit pair/qubit is mapped to its
+     individually best physical edge/qubit; optimizing fidelity; admissible;
+     tight */
+  FidelityBestLocation
+};
+
+/**
+ * A heuristic is admissible if it never overestimates the cost of the best
+ * reachable goal node, i.e. c(n*) <= c(n) + h(n) for cost function c,
+ * heuristic h, any node n in the search graph, and n* the best reachable goal
+ * node from n.
+ */
+[[maybe_unused]] static inline bool isAdmissible(const Heuristic heuristic) {
+  switch (heuristic) {
+  case Heuristic::GateCountMaxDistance:
+  case Heuristic::FidelityBestLocation:
+    return true;
+  case Heuristic::GateCountSumDistanceMinusSharedSwaps:
+  case Heuristic::GateCountMaxDistanceOrSumDistanceMinusSharedSwaps:
+  case Heuristic::GateCountSumDistance:
+    return false;
+  }
+  return false;
```

Comment:
Hm, that's true. I just wonder if it might be better in this specific case to stay explicit. This way, linter warnings for missing switch cases will remind us to keep tracking those properties for new heuristics.

Reply:
Good point 👍🏼 feel free to resolve this then!